overlord/configstate/tasksets.go

```diff
@@ -33,11 +33,12 @@ func init() {
 }
 
 // Configure returns a taskset to apply the given configuration patch.
-func Configure(s *state.State, snapName string, patch map[string]interface{}) *state.TaskSet {
+func Configure(s *state.State, snapName string, patch map[string]interface{}, flags int) *state.TaskSet {
 	hooksup := &hookstate.HookSetup{
-		Snap:     snapName,
-		Hook:     "configure",
-		Optional: len(patch) == 0,
+		Snap:       snapName,
+		Hook:       "configure",
+		Optional:   len(patch) == 0,
+		IgnoreFail: flags&snapstate.IgnoreHookFailure > 0,
```

Comment:
(nitpick land) The `> 0` part is a bit unusual, in C that would be sufficient by itself, perhaps we can just use `!= 0`